addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -1364,73 +1472,96 @@ var VideoWidget = MediaWidget.extend({
-            'hide_yt_logo': this.isForBgVideo || this.$('input#o_video_hide_yt_logo').is(':checked'),
-            'hide_dm_logo': this.isForBgVideo || this.$('input#o_video_hide_dm_logo').is(':checked'),
-            'hide_dm_share': this.isForBgVideo || this.$('input#o_video_hide_dm_share').is(':checked'),
-        });
-
-        var $optBox = this.$('.o_video_dialog_options');
-
-        // Show / Hide preview elements
-        this.$el.find('.o_video_dialog_preview_text, .media_iframe_video_size').add($optBox).toggleClass('d-none', !query.$video);
-        // Toggle validation classes
-        this.$el.find('#o_video_form_group')
-            .toggleClass('o_has_error', !query.$video).find('.form-control, .custom-select').toggleClass('is-invalid', !query.$video)
-            .end()
-            .toggleClass('o_has_success', !!query.$video).find('.form-control, .custom-select').toggleClass('is-valid', !!query.$video);
-
-        // Individually show / hide options base on the video provider
-        $optBox.find('div.o_' + query.platform + '_option').removeClass('d-none');
-
-        // Hide the entire options box if no options are available or if the
-        // dialog is opened for a background-video
-        $optBox.toggleClass('d-none', this.isForBgVideo || $optBox.find('div:not(.d-none)').length === 0);
-
-        if (query.platform === 'youtube') {
-            // Youtube only: If 'hide controls' is checked, hide 'fullscreen'
-            // and 'youtube logo' options too
-            this.$('input#o_video_hide_fullscreen, input#o_video_hide_yt_logo').closest('div').toggleClass('d-none', this.$('input#o_video_hide_controls').is(':checked'));
-        }
-
-        var $content = query.$video;
-        if (!$content) {
-            switch (query.errorCode) {
-                case 0:
-                    $content = $('<div/>', {
-                        class: 'alert alert-danger o_video_dialog_iframe mb-2 mt-2',
-                        text: _t("The provided url is not valid"),
-                    });
-                    break;
-                case 1:
-                    $content = $('<div/>', {
-                        class: 'alert alert-warning o_video_dialog_iframe mb-2 mt-2',
-                        text: _t("The provided url does not reference any supported video"),
-                    });
-                    break;
+        const $textarea = this.$('textarea#o_video_text');
+        const code = $textarea.val().trim();
+        const uploadedVideo = this.$('.o_we_attachment_selected');
+        let query = null;
+        if (uploadedVideo.length && this.selectedAttachments.length) {
+            this._showUpload();
+            query = await this._createVideoNode(this.selectedAttachments[0].src, {
+                'uploaded': true,
+                'autoplay': this.isForBgVideo || this.$('input#o_video_autoplay').is(':checked'),
+                'muted': this.isForBgVideo || this.$('input#o_video_autoplay').is(':checked'),
```

Comment:
Both settings rely on the same input ?

Reply:
Yes, when autoplay is enabled we want the video to be muted as well. Most browsers do not allow unmuted autoplay, so one setting has to decide both (for now).